### PR TITLE
[doc] ROM README.md update for firmware image 2.0 format

### PR DESCRIPTION
**Pull request by mhatrevi**

### CODE REVIEW

File: rom/dev/README.md [hunk 1]
```diff
@@ -83,29 +82,41 @@ Firmware manifest consists of preamble, header and table of contents.
...
-|...<Manufacturer LMS Public Key 32> | | |
-| ECC Public Key Index Hint | 4 | The hint to ROM to indicate which ECC public key it should first use.  |
-| LMS Public Key Index Hint | 4 | The hint to ROM to indicate which LMS public key it should first use.  |
-| Manufacturer ECC Signature | 96 | Manufacturer ECDSA P-384 signature of the Firmware Manifest header hashed using SHA2-384. <br> **R-Coordinate:** Random Point (48 bytes) <br> **S-Coordinate:** Proof (48 bytes) |
-| Manufacturer LMS Signature | 1620 | Manufacturer LMS signature of the Firmware Manifest header hashed using SHA2-384. <br> **q:** Leaf of the Merkle tree where the OTS public key appears (4 bytes) <br> **ots:** Lmots Signature (1252 bytes) <br> **tree_type:** Lms Algorithm Type (4 bytes) <br> **tree_path:** Path through the tree from the leaf associated with the LM-OTS signature to the root. (360 bytes) |
+| Firmware Manifest Type | 4 |  **Byte0:** - Type <br> 0x1 – ECC & LMS Keys <br> 0x2 – ECC & MLDSA Keys <br> **Byte1-Byte3:** Reserved |
```
Comment (swenson) [inline rom/dev/README.md:106]: Are we changing the firmware manifest type as well?

File: rom/dev/README.md [hunk 2]
```diff
@@ -83,29 +82,41 @@ Firmware manifest consists of preamble, header and table of contents.
...
-| Owner LMS Public Key | 48 | LMS public key used to verify the Firmware Manifest Header Signature. <br> **tree_type:** LMS Algorithm Type (4 bytes) <br> **otstype:** LMS Ots Algorithm Type (4 bytes) <br> **id:**  (16 bytes) <br> **digest:**  (24 bytes) |
-| Owner ECC Signature | 96 | Manufacturer ECDSA P-384 signature of the Firmware Manifest header hashed using SHA2-384. <br> **R-Coordinate:** Random Point (48 bytes) <br> **S-Coordinate:** Proof (48 bytes) |
+| Owner LMS or MLDSA Public Key | 48 or 2592 | LMS public key used to verify the Firmware Manifest Header Signature. <br> **tree_type:** LMS Algorithm Type (4 bytes) <br> **otstype:** LMS Ots Algorithm Type (4 bytes) <br> **id:**  (16 bytes) <br> **digest:**  (24 bytes) <br><br>**OR**<br><br>MLDSA-87 public key used to verify the Firmware Manifest Header Signature. <br> (2592 bytes)|
+| Owner ECC Signature | 96 | Manufacturer ECC P-384 signature of the Firmware Manifest header hashed using SHA2-384. <br> **R-Coordinate:** Random Point (48 bytes) <br> **S-Coordinate:** Proof (48 bytes) |
```
Comment (swenson) [inline rom/dev/README.md:119]: Don't we need to update the Owner signature as well for MLDSA?

File: rom/dev/README.md [hunk 3]
```diff
@@ -584,28 +595,27 @@ The following are the pre-conditions that should be satisfied:
 ## Preamble validation: Validate the manufacturing keys
 
 - Load the preamble bytes from the mailbox.
-- There are four ECC and thirty-two LMS manufacturing keys in the preamble.
-- fuse_key_manifest_pk_hash is the fuse that contains the hash of all the ECC and LMS manufacturing keys.
-- To validate the key region, take the hash of all the ECC and LMS keys and compare it against the hash in fuse.
+- There is an ECC key descriptor and either LMS or MLDSA key descriptor in the preamble. The ECC descriptor contains upto four ECC public key hashes. The LMS key descriptor contains upto 32 public key hashes. The MLDSA key descriptor contains upto four MLDSA public key hashes.
```
Comment (swenson) [inline rom/dev/README.md:598]:
```suggestion
- There is an ECC key descriptor and either LMS or MLDSA key descriptor in the preamble. The ECC descriptor contains up to four ECC public key hashes. The LMS key descriptor contains up to 32 public key hashes. The MLDSA key descriptor contains up to four MLDSA public key hashes.
```

File: rom/dev/README.md (hunk 1 above)
Comment (mhatrevi) [reply rom/dev/README.md:106]: This is a new field we are adding.

File: rom/dev/README.md (hunk 2 above)
Comment (mhatrevi) [reply rom/dev/README.md:119]: Fixed.

File: rom/dev/README.md (hunk 3 above)
Comment (mhatrevi) [reply rom/dev/README.md:598]: Fixed

**Review by swenson — DISMISSED** [review]
LGTM

We had discussed changing the firmware manifest marker (since we don't have a version) so that it can't be confused with the v1 firmware manifest. Did we decide to do that?

**Review by swenson — APPROVED** [review]

**Pull request merged**